source_handlers.go

```diff
@@ -149,7 +149,57 @@ func SourceEdit(c echo.Context) error {
 		return err
 	}
 
-	s.UpdateFromRequest(input)
+	if isPaused {
+		binder := SourcePausedEditBinder{}
```

Comment:
I agree this puts a weird taste in my mouth. We also wouldn't want to use a binder by itself here since we can theoretically only have one binder at the echo router level. I see this as more of a validation in a service class if we did go this route. 

----

fwiw I would rather we have a `SourcePauseEdit` struct and let the "regular" binder return an error, and return a 400 that way. I'm not sure the multi-response is necessary since 99.9% of the time the user is just updating availability status.